Library/Homebrew/requirements/macos_requirement.rb

```diff
@@ -59,4 +59,10 @@ def display_s
 
     "macOS #{@comparator} #{@version}"
   end
+
+  def to_json(*args)
+    {
+      @comparator.to_s => (@version.is_a?(Array) ? @version.map(&:to_s) : [@version.to_s]),
```

Comment:
Can you split this into an `if`/`else` and use some intermediate variables? It's pretty hard to follow as-is. Thanks!

Reply:
Have a look. ↓